### PR TITLE
test: add new test cases for pkg/protocol/response.go

### PR DESCRIPTION
**Pull request by gityh2021**
#### What type of PR is this?
test

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
测试：为pkg/protocol/response.go文件新增测试用例。

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: add test cases for the following features for pkg/protocol/response.go and the test coverage rate of pkg/protocol/response.go is improved to 81.08%.
      1.TestResponseMustSkipBody：test when to skip body read and copy.
      2.TestResponseBodyGunzip: test compress and decompress body with gzip.
      3.TestResponseSwapResponseBody.
      4.TestResponseAcquireResponse: test acquire and release response.
zh(optional):

#### Which issue(s) this PR fixes:
https://github.com/cloudwego/hertz/issues/257

### CODE REVIEW

**Comment by L2ncE** [comment]
You can improve your description according to the PR template and can describe the situation of unit test coverage.

**Comment by welkeyever** [comment]
Please resign the CLA🙏

**Comment by gityh2021** [comment]
> 

done

> Please resign the CLA🙏

done

**Comment by welkeyever** [comment]
It is best to note the change in overall coverage in the description.

File: pkg/protocol/response_test.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/cloudwego/hertz/pkg/common/bytebufferpool"
+	"github.com/cloudwego/hertz/pkg/common/compress"
+
```
Comment (welkeyever) [inline pkg/protocol/response_test.go:53]: no need this blank line here

**Comment by gityh2021** [comment]
> It is best to note the change in overall coverage in the description.

May I use the the test coverage rate

**Comment by gityh2021** [comment]
> It is best to note the change in overall coverage in the description.

done, sorry to close this pr...

File: pkg/protocol/response_test.go (hunk 1 above)
Comment (gityh2021) [reply pkg/protocol/response_test.go]: done, thanks~

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**